vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridSingleSelectionIT.java

```diff
@@ -85,4 +85,25 @@ public void checkDeselectionDisallowedInitially() {
         Assert.assertTrue("Row 1 was still selected after de-selecting it.",
                 !grid.getRow(1).isSelected());
     }
+
+    @Test
+    public void gridSetItemsAndSelectItemAndSetItemAfterSetDeselectAllowedFalse() {
```

Comment:
This test case doesn't seem to be valid. The build passes fine without the changes to the connector.

Reply:
after discussed, I corrected the test and it covers the bug